### PR TITLE
Update README (for env vars)

### LINKED ISSUE #12: Update README with correct env vars
Add `OTEAPI_` prefix to all env vars for connecting with redis.

### LINKED ISSUE #13: Consider switching the order of `development` and `production` in Dockerfile
This is to avoid all the testing being performed when building the `production` target.
It seems this is not automatically avoided as the build process reads line by line until it gets a satisfactory target followed by a `CMD` or `ENTRYPOINT` command.

### PR DESCRIPTION
**Pull request by CasperWA**
Fixes #12 
~Closes #13~

~It is noted that with the new target order, no tests will be run for the `production` target. Neither during build or during run.
Before, tests would be run during the build, but not when running.~

~There are no changes for `development`.~

Update README.

### CODE REVIEW

**Comment by francescalb** [comment]
I still get this: 


=========================== short test summary info ============================
ERROR tests/routers/test_filter.py::test_create_filter - KeyError: "index=('s...
ERROR tests/routers/test_filter.py::test_get_filter - KeyError: "index=('sche...
ERROR tests/routers/test_filter.py::test_initialize_filter - KeyError: "index...
ERROR tests/routers/test_mapping.py::test_create_mapping - KeyError: "index=(...
ERROR tests/routers/test_mapping.py::test_get_mapping - KeyError: "index=('sc...
ERROR tests/routers/test_mapping.py::test_initialize_mapping - KeyError: "ind...
ERROR tests/routers/test_session.py::test_list_session - KeyError: "index=('s...
ERROR tests/routers/test_session.py::test_create_session - KeyError: "index=(...
ERROR tests/routers/test_transformation.py::test_create_transformation - KeyE...
ERROR tests/routers/test_transformation.py::test_get_transformation - KeyErro...
ERROR tests/routers/test_transformation.py::test_initialize_transformation - ...
ERROR tests/routers/test_transformation.py::test_get_transformation_status - ...
ERROR tests/routers/test_transformation.py::test_execute_transformation - Key...
=================== 1 passed, 2 warnings, 13 errors in 7.22s ===================
Error: error building at STEP "RUN pytest --cov app": error while running runtime: exit status 1

**Comment by francescalb** [comment]
> I still get this:
> 
> =========================== short test summary info ============================ ERROR tests/routers/test_filter.py::test_create_filter - KeyError: "index=('s... ERROR tests/routers/test_filter.py::test_get_filter - KeyError: "index=('sche... ERROR tests/routers/test_filter.py::test_initialize_filter - KeyError: "index... ERROR tests/routers/test_mapping.py::test_create_mapping - KeyError: "index=(... ERROR tests/routers/test_mapping.py::test_get_mapping - KeyError: "index=('sc... ERROR tests/routers/test_mapping.py::test_initialize_mapping - KeyError: "ind... ERROR tests/routers/test_session.py::test_list_session - KeyError: "index=('s... ERROR tests/routers/test_session.py::test_create_session - KeyError: "index=(... ERROR tests/routers/test_transformation.py::test_create_transformation - KeyE... ERROR tests/routers/test_transformation.py::test_get_transformation - KeyErro... ERROR tests/routers/test_transformation.py::test_initialize_transformation - ... ERROR tests/routers/test_transformation.py::test_get_transformation_status - ... ERROR tests/routers/test_transformation.py::test_execute_transformation - Key... =================== 1 passed, 2 warnings, 13 errors in 7.22s =================== Error: error building at STEP "RUN pytest --cov app": error while running runtime: exit status 1

When trying to run development mode

**Comment by francescalb** [comment]
SAme in production mode: 
Error: error building at STEP "RUN pytest --cov app": error while running runtime: exit status 1

**Review by francescalb — CHANGES_REQUESTED** [review]
Add a  ´--target production´ to the readme on how to build image for production.

**Comment by CasperWA** [comment]
The target order in the Dockerfile should not be changed, as we want to run the tests when building the `production` target image. The changes related to this order change have been reverted.

**Comment by CasperWA** [comment]
@francescalb Can this be approved (and merged) :)

File: README.md [hunk 1]
```diff
@@ -24,7 +24,9 @@ To build and run the production target from the command line:
 ```shell
 docker build --rm -q -f Dockerfile \
     --label "ontotrans.oteapi=production" \
+    --target production \
```
Comment (francescalb) [inline README.md:27]: This is not needed anymore since the order has been changed back.

Comment (CasperWA) [reply README.md:27]: It makes it explicit however? But you're right - it's not needed as it will be the default now. Do you want to remove it or keep it?

**Comment by jesper-friis** [comment]
These instructions works well for me with docker. Like Francesca I have some problems with podman (which I don't like to run as root).

A few general suggestions for the README:
- Remove the `-q` option to `docker build`.
- It is little confusing that docker-compose listen by default to port 80, while the explicit commands in the README listen to port 8080. I will suggest to change the default exported port in docker-compose to 8080. That may also help with running podman in rootless mode, which doesn't allow to export privileged ports. 
- For the SFTP commands, we could replace `/home/foo/upload` with `${HOME}/Download` and `foo:pass` with `${USER}:${PASSWORD}`. Instead of explicit password, ssh public key authentication is preferred.

**Comment by CasperWA** [comment]
> A few general suggestions for the README:
> 
> * Remove the `-q` option to `docker build`.

👍 

> * It is little confusing that docker-compose listen by default to port 80, while the explicit commands in the README listen to port 8080. I will suggest to change the default exported port in docker-compose to 8080. That may also help with running podman in rootless mode, which doesn't allow to export privileged ports.

The point of docker-compose is to be more of a template/example of how one would do it in a production environment. It's not meant for development or testing, really. So I'm slightly against changing to another port.
However, it can still be justified, by simply remarking that one would most likely be running a reverse proxy (nginx or similar), which will map to some internal port where the OTE API service is running - so instead of 80 it would be 8531 or something custom :) So sure, 👍 

But this should probably be a yellow flag that we need some more documentation on the purposes of docker-compose and Dockerfile.

> * For the SFTP commands, we could replace `/home/foo/upload` with `${HOME}/Download` and `foo:pass` with `${USER}:${PASSWORD}`. Instead of explicit password, ssh public key authentication is preferred.

Sure.
There is some Windows/Unix issues in those paths and env vars as well, but sure.

File: README.md (hunk 1 above)
Comment (francescalb) [reply README.md:27]: It can be kept.

**Review by jesper-friis — APPROVED** [review]
I am happy. Thanks for the changes

**Pull request merged**